app/decorators/models/course_decorator.rb

```diff
@@ -144,12 +144,20 @@ def coalesce_datetime(date:, time:)
   end
 
   def utc_day_offset(time)
-    parsed_in_zone = Time.parse(time)
+    parsed_in_zone = parse_time_in_zone(time)
     hour_offset = (parsed_in_zone.utc_offset / 3_600) * -1
     rollover_limit = 24 - hour_offset
     parsed_in_zone.hour >= rollover_limit ? 1 : 0
   end
 
+  def parse_time_in_zone(time)
+    school_timezone = SettingsService.get_settings(object: 'school', id: 1)['timezone']
+    custom_timezones = {
+      'MT' => ActiveSupport::TimeZone['America/Denver'],
```

Comment:
Is this problem something that is actively only affected the MT timezone, or would it be affecting our DLS oberseving timezones?

Reply:
We believe that could be the case, but we have decided to take the problems as they come. If we have issues with other timezones, we should add them to custom_timezones.